pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -525,6 +529,20 @@ func shouldH2Upgrade(clusterName string, direction model.TrafficDirection, port
 		return false
 	}
 
+	// TODO (mjog)
+	// Upgrade if tls.GetMode() == networking.TLSSettings_ISTIO_MUTUAL
+	override := networking.ConnectionPoolSettings_HTTPSettings_DEFAULT
+	if connectionPool != nil && connectionPool.Http != nil {
+		override = connectionPool.Http.H2UpgradePolicy
+	}
+	// If user wants an upgrade at destination rule/port level that means he is sure that
+	// it is a Http port - upgrade in such case. This is useful incase protocol sniffing is
+	// enabled and user wants to upgrade/preserve http protocol from client.
+	if override == networking.ConnectionPoolSettings_HTTPSettings_UPGRADE {
```

Comment:
I think We should only upgrade if connectionPool.http is specified, and not if global default is upgrade.

Reply:
Yes. That is what `override` is. It the one set in DR's `connectionPool.http`